### PR TITLE
feat: add Reversal() to files

### LINKED ISSUE #756: file: offer a Reversal method
**Problem Describe**
If a duplicate or erroneous file was sent off to an FI it can be corrected by sending a reversal ACH file. This reversal has a specific transform from the original. These definitions come from the NACHA guidelines, but the process to create a reversal file is below.

Formatting Requirements for Reversing Entries:

> Must submit a Reversing Entry within a batch that includes the word “REVERSAL” in the Company Entry Description field of the Company/Batch Header Record.  The description “REVERSAL” must replace the original content of the Company Entry Description field transmitted in the original batch, including content that may otherwise have been required by The Rules.
> For each Reversing Entry, the content on the following fields must remain unchanged from the original, erroneous Entry to which the Reversal relates: 
>  - Standard Entry Class Code
>  - Company Identification/Originator Identification
>  - Amount

> Update the following records according to the fund flow
>   - Update TransactionCode fields. 
>   - Set EffectiveEntryDate properly (for same-day vs standard) 

> The name of the Originator (as identified in the Company Name field or the Originator Name field) must reflect the same Originator identified in the Erroneous Entry to which the Reversal relates.

### PR DESCRIPTION
**Pull request by adamdecaf**
Closes: https://github.com/moov-io/ach/issues/756

### CODE REVIEW

File: reversal.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// under the License.
+
+package ach
+
+import (
+	"time"
+)
+
+// Reversal will transform a File into a Nacha compliant reversal which can be transmitted to undo fund movement.
+func (f *File) Reversal(effectiveEntryDate time.Time) error {
```
Comment (atonks2) [inline reversal.go:25]: Would it be worth checking here `if effectiveEntryDate.IsZero()` and returning an error?

Comment (adamdecaf) [reply reversal.go:25]: Perhaps. That'll fail validation as `000000` isn't a valid EffectiveEntryDate value.

File: reversal.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		// Adjust Effective Entry Date for same-day vs standard
+		bh.EffectiveEntryDate = effectiveEntryDate.Format("060102")
+
+		f.Batches[i].SetHeader(bh)
+
+		// In EntryDetail records we need to update the TransactionCode fields to undo fund movement.
+		entries := f.Batches[i].GetEntries()
+		for j := range entries {
+			// 22 (CheckingCredit) -> 27 (CheckingDebit)
+			entries[j].TransactionCode += 5
```
Comment (atonks2) [inline reversal.go]: Is this always going to end up as `27`?, or is `TransactionCode + 5` always the correct code for a reversal?

Comment (adamdecaf) [reply reversal.go]: There are a few ranges of these codes, but it's correct to `+= 5` them all. 

https://github.com/moov-io/ach/blob/v1.21.2/entryDetail.go#L112

Comment (atonks2) [reply reversal.go]: Well that's convenient. Thanks!

**Review by atonks2 — DISMISSED** [review]

File: reversal.go (hunk 2 above)
Comment (adamdecaf) [reply reversal.go]: There's a bug with 27 -> 32 (which is incorrect).

**Review by atonks2 — DISMISSED** [review]

**Pull request merged**